File: backend/candidate-profile-service/src/main/java/com/jobportal/candidateprofile/entities/Skill.java
package com.jobportal.candidateprofile.entities;

import jakarta.persistence.*;

@Entity
@Table(name = "skill_table")
public class Skill {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "skillid")
    private Integer skillId;

    @Column(name = "skillname", nullable = false, unique = true, length = 100)
    private String skillName;

    public Skill() {
    }

    public Skill(Integer skillId, String skillName) {
        this.skillId = skillId;
        this.skillName = skillName;
    }

    public Integer getSkillId() {
        return skillId;
    }

    public void setSkillId(Integer skillId) {
        this.skillId = skillId;
    }

    public String getSkillName() {
        return skillName;
    }

    public void setSkillName(String skillName) {
        this.skillName = skillName;
    }
}

File: backend-dotnet/BuildingBlocks/JobPortal.Shared/Entities/Skill.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobPortal.Shared.Entities;

[Table("skill_table")]
public class Skill
{
    [Key]
    [Column("skillid")]
    public int SkillId { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("skillname")]
    public string SkillName { get; set; } = string.Empty;
}
